### PR TITLE
Changing README with new Session Management Changes

### PR DESCRIPTION
**Pull request by sojanpr**
Added Auto Session Management details @qinweigong @aaustin

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -212,7 +212,51 @@ Branch branch = Branch.getInstance(getApplicationContext(), "your test branch ke
 Either way, we recommend you put a `//TODO` to remind you to change back to live app during deployment later. 
 Also, note the Branch object is singleton, so you can and should still use `Branch.getInstance(getApplicationContext())` in all the other places (see examples below).
 
+#### Automatic Session Management
+Starting from 1.5.7, there is no need for intialising and closing session. Branch SDK can do it for you if you do one of the following options.
+
+1) If you are not creating an Application class, declare ```BranchApp``` as your application class in your manifest.
```
Comment (qinweigong) [inline README.md]: For inline code, you can use single back-tick instead of three, e.g. `BranchApp`

File: README.md [hunk 2]
```diff
@@ -212,7 +212,51 @@ Branch branch = Branch.getInstance(getApplicationContext(), "your test branch ke
 Either way, we recommend you put a `//TODO` to remind you to change back to live app during deployment later. 
 Also, note the Branch object is singleton, so you can and should still use `Branch.getInstance(getApplicationContext())` in all the other places (see examples below).
 
+#### Automatic Session Management
+Starting from 1.5.7, there is no need for intialising and closing session. Branch SDK can do it for you if you do one of the following options.
```
Comment (qinweigong) [inline README.md]: You may want to mention the android version requirement here!

File: README.md [hunk 3]
```diff
@@ -212,7 +212,51 @@ Branch branch = Branch.getInstance(getApplicationContext(), "your test branch ke
...
+android:name="io.branch.referal.BranchApp">
+```
+
+2) If you already have an Application class then extend your application class with ```BranchApp```.
+
+```java
+public class YourApplication extends BranchApp
+```
+
+3) If you already have an Application class and don't want to extend it from ```BranchApp``` then create an Branch instance in your Application's ```onCreate()``` method.
```
Comment (qinweigong) [inline README.md]: an -> a

File: README.md [hunk 4]
```diff
@@ -212,7 +212,51 @@ Branch branch = Branch.getInstance(getApplicationContext(), "your test branch ke
...
+	}
+}
+```
+
+Once you do any of the above, no need to close or init sessions in your Activities. Branch SDK will do all that for you. You can get your Branch instance at any time as follows.
+
+```java
+  Branch branch = Branch.getInstance();
+```
+Automatic Session Management can work only with API level 14 and above. So make sure that your minSdkVersion is 14 or above.
```
Comment (qinweigong) [inline README.md]: Ahh, it's here!

File: README.md [hunk 5]
```diff
@@ -212,7 +212,51 @@ Branch branch = Branch.getInstance(getApplicationContext(), "your test branch ke
...
+
+```xml
+ <uses-sdk
+ 	android:minSdkVersion="14"
+        ------------        />
+```
+
 #### Close session
+Note:Skip this if you are implementing Automatic Session Management as described above
```
Comment (qinweigong) [inline README.md]: You can make "above" a link to the section.

File: README.md (hunk 2 above)
Comment (qinweigong) [inline README.md]: Ahh, now I regret. I should've suggested you to bump the version to 1.6.0, given it's a big change.

File: README.md [hunk 6]
```diff
@@ -209,10 +209,57 @@ Or
...
+
+1) If you are not creating an Application class, declare `BranchApp` as your application class in your manifest.
+
+```xml
+ <application
+-----
+android:name="io.branch.referal.BranchApp">
+```
+
+2) If you already have an Application class then extend your application class with `BranchApp`.
```
Comment (aaustin) [inline README.md]: This is not really 1 and 2 or 3 for steps. These should be ORs - let me pull down and modify layout.

**Comment by aaustin** [comment]
:+1:

**Pull request merged**